python/templates/index.ejs

```diff
@@ -235,27 +262,69 @@
       $('#assets').show();
     }
 
-    var handler = Plaid.create({
-      apiVersion: 'v2',
-      clientName: 'Plaid Quickstart',
-      env: '{{ plaid_environment }}',
-      product: products,
-      key: '{{ plaid_public_key }}',
-      countryCodes: '{{ plaid_country_codes }}'.split(','),
-      // webhook: 'https://your-domain.tld/plaid-webhook',
-      onSuccess: function(public_token) {
-        $.post('/get_access_token', {
-          public_token: public_token
-        }, function(data) {
-          $('#container').fadeOut('fast', function() {
-            $('#item_id').text(data.item_id);
-            $('#access_token').text(data.access_token);
-            $('#intro').hide();
-            $('#app, #steps').fadeIn('slow');
-          });
+    // This functionality is only relevant for the UK Payment Initiation product.
+    if (products.includes('payment_initiation')) {
+      $('.payment_initiation').show();
+      $.post('/set_payment_token', {}, function(data) {
+        var paymentToken = data.payment_token;
+        $('#payment_token').text(paymentToken);
+
+        // In the case of payment_initiation product, we need to wait for
+        // payment token to be generated before the Link handler can be
+        // initialized.
+        handler = Plaid.create({
+          apiVersion: 'v2',
+          clientName: 'Plaid Quickstart',
+          env: '{{ plaid_environment }}',
+          product: products,
+          key: '{{ plaid_public_key }}',
+          countryCodes: '{{ plaid_country_codes }}'.split(','),
+          paymentToken: paymentToken,
+          language: 'en',
+          onSuccess: function(public_token) {
+            // This public token exchange step is not relevant for the
+            // payment_initiation product and should be skipped.
+            $.post('/get_access_token', {
```

Comment:
It looks like this call to `$.post('/get_access_token'` isn't actually needed based on the comment?

Reply:
Yes but it is required for the success page to show item id, access token, etc so I did not want to break that.